metadata-ingestion/src/datahub/ingestion/source/bigquery/bigquery_audit.py

```diff
@@ -0,0 +1,573 @@
+                OR
+                "google.cloud.bigquery.v2.JobService.InsertJob"
+            )
+        AND
+        protoPayload.metadata.jobChange.job.jobStatus.jobState="DONE"
+        AND NOT protoPayload.metadata.jobChange.job.jobStatus.errorResult:*
+        AND protoPayload.metadata.jobChange.job.jobStats.queryStats.referencedTables:*
+        AND NOT protoPayload.metadata.jobChange.job.jobStats.queryStats.referencedTables =~ "projects/.*/datasets/.*/tables/__TABLES__|__TABLES_SUMMARY__|INFORMATION_SCHEMA.*"
+         AND (
+            {allow_regex}
+            {deny_regex}
+         OR
+            protoPayload.metadata.tableDataRead.reason = "JOB"
+        )
+    )
+    OR
+    (
+        protoPayload.metadata.tableDataRead:*
+    )
+)
+AND
+timestamp >= "{start_time}"
+AND
+timestamp < "{end_time}"
+""".strip(
+        "\t \n"
+    ),
+}
+
+DEBUG_INCLUDE_FULL_PAYLOADS = False
+
+AuditLogEntry = Any
+
+# BigQueryAuditMetadata is the v2 format in which audit logs are exported to BigQuery
+BigQueryAuditMetadata = Any
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _make_gcp_logging_client(
+    project_id: Optional[str] = None, extra_client_options: Dict[str, Any] = {}
+) -> GCPLoggingClient:
+    # See https://github.com/googleapis/google-cloud-python/issues/2674 for
+    # why we disable gRPC here.
+    client_options = extra_client_options.copy()
+    client_options["_use_grpc"] = False
+    if project_id is not None:
+        return GCPLoggingClient(**client_options, project=project_id)
+    else:
+        return GCPLoggingClient(**client_options)
+
+
+@dataclass(frozen=True, order=True)
+class BigqueryTableIdentifier:
+    project_id: str
+    dataset: str
+    table: str
+
+    _BIGQUERY_DEFAULT_SHARDED_TABLE_REGEX: str = "((.+)[_$])?(\\d{4,10})$"
+
+    def _get_table_and_shard(self, table_name: str) -> Tuple[str, Optional[str]]:
+        match = re.search(
+            BigqueryTableIdentifier._BIGQUERY_DEFAULT_SHARDED_TABLE_REGEX,
+            table_name,
+            re.IGNORECASE,
+        )
+        if match:
+            table_name = match.group(2)
+            shard = match.group(3)
+            return table_name, shard
+        return table_name, None
+
+    @classmethod
+    def from_string_name(cls, table: str) -> "BigqueryTableIdentifier":
+        parts = table.split(".")
+        return cls(parts[0], parts[1], parts[2])
+
+    def raw_table(self):
+        return f"{self.project_id}.{self.dataset}.{self.table}"
+
+    @staticmethod
+    def _remove_suffix(input_string, suffix):
+        if suffix and input_string.endswith(suffix):
+            return input_string[: -len(suffix)]
+        return input_string
+
+    def get_table(self) -> str:
+        shortened_table_name = self.table
+        # if table name ends in _* or * then we strip it as that represents a query on a sharded table
+        shortened_table_name = self._remove_suffix(shortened_table_name, "_*")
```

Comment:
pass a list or better yet, a regex pattern to the `_remove_suffix` instead of multiple calls here?